hexrd/unitcell.py

```diff
@@ -1262,94 +1269,130 @@ def compliance(self, v):
     # lattice constants as properties
 
     @property
-    def a(self):
-        return self._a
+    def lparms(self):
+        return [self.a, self.b,
+                self.c, self.alpha, self.beta,
+                self.gamma]
 
-    @a.setter
-    def a(self, val):
-        self._a = val
+    @lparms.setter
+    def lparms(self, lp):
+        """
+        set the lattice parameters here
+        """
+        self._a = lp[0].getVal("nm")
+        self._b = lp[1].getVal("nm")
+        self._c = lp[2].getVal("nm")
+        self._alpha = lp[3].getVal("degrees")
+        self._beta = lp[4].getVal("degrees")
+        self._gamma = lp[5].getVal("degrees")
         self.calcmatrices()
-        self.ih = 1
-        self.ik = 1
-        self.il = 1
+        self.init_max_g_index()
         self.CalcMaxGIndex()
         if(hasattr(self, 'numat')):
             self.CalcDensity()
 
+    @property
+    def lparms_reduced(self):
+        lp = self.lparms
+        lp_red = [lp[i] for i in
+                  _rqpDict[self.latticeType][0]]
+        return lp_red
+
+    @property
+    def a(self):
+        return self._a
+
+    @a.setter
+    def a(self, val):
+        if self.is_editable("a"):
+            lp = self.lparms
+            lp_valunit = self.fill_correct_lp_vals(
+                lp, val, "a")
+            self.lparms = lp_valunit
+        else:
+            msg = (f"not an editable field"
+                   f" for this space group")
+            print(msg)
```

Comment:
It might be better if these are exceptions rather than just a print statement.

Reply:
ok modified to a `RuntimeError`